### PR TITLE
Tests for claims processing + Code for handling tar.gz input

### PR DESCRIPTION
**Pull request by itsmemala**
1. Added test cases for claims processing and made any associated code updates.
2. The test case for the scispacy dependent function is currently commented out because it runs into a "fork:cannot allocate memory" error on Travis.
3. Updated cord pre-processing code to handle tar.gz type input for "document_parses" directory

### CODE REVIEW

File: src/contradictory_claims/data/process_claims.py
Comment (dnsosa) [inline src/contradictory_claims/data/process_claims.py:118]: For a given claim, are the drug term mentions coming from the entire paper, from any claim in that paper, or from that specific claim? I ask because I did one additional filter to check that the claim1 and claim2 text have a common drug mention, where I'm looking for a drug mention in a custom regex. And iirc, I was filtering out some claim pairs where the two claims don't have a common drugs mentioned in the text (although they might have an overlap in the drug_mentions column).

**Review by dnsosa — COMMENTED** [review]
Great! Love all the testing. Just had a couple of minor comments/questions to check on

File: src/contradictory_claims/cli.py
Comment (itsmemala) [reply src/contradictory_claims/cli.py:39]: Yeah makes sense, let's merge this after merging yours. I'll make the cord-preprocessing->claims extraction steps

File: tests/data/constants.py
Comment (itsmemala) [reply tests/data/constants.py:16]: Yeah, I think it's fine to exclude the date form the testing dir structure so it's easier to swap files without changes to the constants.py file each time.

File: tests/data/constants.py
Comment (itsmemala) [reply tests/data/constants.py:14]: No, only the paths starting with "sample_" are in "test/data/resources" and are actually being imported for the tests. I just kept the original input paths in there as well, following the format you'd used. Maybe they're there for reference?

File: src/contradictory_claims/data/process_claims.py
Comment (itsmemala) [reply src/contradictory_claims/data/process_claims.py:118]: So originally in Michael's code, the drug_terms_mentions column had the drugs present in the section from which the claims were extracted (comment in line #107), and the processing of earlier versions (including v31 , if I'm not wrong) were based on this.
While refactoring the code, I added this step to (line #118) so that when pairing, we're only comparing the drug names present in the claim sentence. This should reflect in the latest extraction of the 2020-08-15 cord version.

**Review by dnsosa — APPROVED** [review]

**Comment by dnsosa** [comment]
NOTE: I merged a commit to resolve some merge conflicts from this branch. I think I resolved the pre-process CORD ones correctly, but if not this is a good spot to look if there was a breaking change

**Comment by dnsosa** [comment]
@itsmemala can you check out the Travis issue? Looks like json_normalize is maybe never defined in preprocess_cord?

**Comment by itsmemala** [comment]
> @itsmemala can you check out the Travis issue? Looks like json_normalize is maybe never defined in preprocess_cord?

It seems like this is resulting from the merge done for resolving the merge conflicts. The import statement does exist in my branch, but I think if I push that file it will lead to the merge conflict again? Shall I push it anyway and we can try to resolve any merge conflicts properly this time?

**Pull request merged**
